File: 剑指Offer/链表中倒数k个节点/listk.java
/*
public class ListNode {
    int val;
    ListNode next = null;
 
    ListNode(int val) {
        this.val = val;
    }
}*/
public class Solution {
    public ListNode FindKthToTail(ListNode head,int k) {
        ListNode right = head;
        ListNode left = head;
        int i = 0;
        while(i<k && right!=null){
            right = right.next;
            i++;
        }
        if(i<k){
            return null;
        }
        while(right!=null){
            right = right.next;
            left = left.next;
        }
        return left;
    }
}

File: 剑指Offer/链表中倒数k个节点/listk.py
# -*- coding:utf-8 -*-
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
 
class Solution:
    def FindKthToTail(self, head, k):
        # write code here
        left,right,i = head,head,0
        while i<k and right:
            right = right.next
            i+=1
        if i<k:
            return None
        while right:
            left = left.next
            right = right.next
        return left